uberpoet/cpprojectgen.py

```diff
@@ -142,17 +171,32 @@ def gen_app_main(self, app_node, module_index):
 
     # Library Generation
 
-    def gen_lib_module(self, module_node, loc_per_unit):
+    def gen_lib_module(self, module_node, loc_per_unit, language):
         # Make Podspec Text
         deps = self.make_dep_list([i.name for i in module_node.deps])
         pod_text = self.pod_lib_template.format(module_node.name, deps, self.wmo_state)
 
-        # Make Swift Text
-        file_count = (max(self.swift_file_size_loc, loc_per_unit) * module_node.code_units) / self.swift_file_size_loc
-        if file_count < 1:
-            raise ValueError("Lines of code count is too small for the module {} to fit one file, increase it.".format(
-                module_node.name))
-        files = {"File{}.swift".format(i): self.swift_gen.gen_file(3, 3) for i in xrange(file_count)}
+        # Make Text
+        if language == Language.SWIFT:
+            file_count = (
+                max(self.swift_file_size_loc, loc_per_unit) * module_node.code_units) / self.swift_file_size_loc
+            if file_count < 1:
+                raise ValueError(
+                    "Lines of code count is too small for the module {} to fit one file, increase it.".format(
+                        module_node.name))
+            files = {"File{}.swift".format(i): self.swift_gen.gen_file(3, 3) for i in xrange(file_count)}
+        elif language == Language.OBJC:
+            file_count = (max(self.objc_file_size_loc, loc_per_unit) * module_node.code_units) / self.objc_file_size_loc
+            if file_count < 1:
+                raise ValueError(
+                    "Lines of code count is too small for the module {} to fit one file, increase it.".format(
+                        module_node.name))
+            files = {}
```

Comment:
I am not in love with this either, should we just make `objc_file_gen` to produce both files and instead return an array of files that were produced? Similarly we can update the Swift one to just return an array of a single Swift file it generated.

Reply:
What do you think ?

Reply:
> but I think we should move out swift & objc codegen into it's own generalized class or function, that way bazel codegen can also just use it verbatim. It's gotten a lot more complicated from the original version.

Agree! I thought the same. Will most likely extract it.